internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -93,6 +97,11 @@ func New(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOpti
 	r := &delegatingResolver{
 		target: target,
 		cc:     cc,
+		// Child resolvers may send a state update as soon as they're built.
+		// Initialize children with no-op resolvers. When both children are nil,
+		// the delegatingResolver is considered closed.
```

Comment:
Related: do we not need to hold `childMu` when calling `Build`?

Reply:
Yes, we need to. Updates from one child can trigger ResolveNow calls in the other child. This also avoids the need to initialize children here. Added code to acquire the mutex and release it.